components/builder-core/src/rpc.rs

```diff
@@ -66,14 +69,12 @@ impl RpcClient {
     pub fn new(url: &str) -> Self {
         debug!("Creating RPC client, url = {}", url);
 
-        let mut headers = Headers::new();
-        headers.set(Accept(vec![qitem(mime::APPLICATION_JSON)]));
-        headers.set(ContentType::json());
-
-        let mut cli = Client::builder();
-        cli.default_headers(headers);
+        let mut headers = HeaderMap::new();
+        headers.insert(USER_AGENT, HeaderValue::from_static(BLDR_USER_AGENT));
+        headers.insert(ACCEPT, HeaderValue::from_static(APPLICATION_JSON));
+        headers.insert(CONTENT_TYPE, HeaderValue::from_static(APPLICATION_JSON));
 
-        RpcClient { cli:      cli.build().unwrap(),
+        RpcClient { cli:      Client::builder().default_headers(headers).build().unwrap(),
```

Comment:
Similar to elsewhere, [this can fail due to TLS errors,](https://docs.rs/reqwest/0.9.18/reqwest/struct.ClientBuilder.html#errors) and I think it may be better to propagate that error than panic (or at least print out the error details before panicking).

Reply:
Changing to an expect